### ISSUE TITLE
Strtolower not inferring type of literals

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#8540]
https://psalm.dev/r/a2fea81889

About this, I've been thinking of resurrecting my literal inferring PR after I'm done with immutability :)

**Comment by orklah** [comment]
yeah, I'm fine with adding this, even if the usefulness is quite limited from my point of view (I get literal arithmetic operations, for things like `$var = 2 * 1024 * 1024;` for readability, but I'm more skeptic about people having literal mixed cases strings they then pass through strtolower and for which the result is somehow relevant for a further check that need to know exactly that lowercased string)

**Comment by danog** [comment]
In my usecase it's a typed property that takes two lowercased literal values
